specification/storage/resource-manager/Microsoft.Storage/stable/2021-04-01/storage.json

```diff
@@ -0,0 +1,4549 @@
+          "description": "An object that defines the filter set."
+        },
+        "format": {
+          "type": "string",
+          "description": "This is a required field, it specifies the format for the inventory files.",
+          "enum": [
+            "Csv",
+            "Parquet"
+          ],
+          "x-ms-enum": {
+            "name": "format",
+            "modelAsString": true
+          }
+        },
+        "schedule": {
+          "type": "string",
+          "description": "This is a required field. This field is used to schedule an inventory formation.",
+          "enum": [
+            "Daily",
+            "Weekly"
+          ],
+          "x-ms-enum": {
+            "name": "schedule",
+            "modelAsString": true
+          }
+        },
+        "objectType": {
+          "type": "string",
+          "description": "This is a required field. This field specifies the scope of the inventory created either at the blob or container level.",
+          "enum": [
+            "Blob",
+            "Container"
+          ],
+          "x-ms-enum": {
+            "name": "objectType",
+            "modelAsString": true
+          }
+        },
+        "schemaFields": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "This is a required field. This field specifies the fields and properties of the object to be included in the inventory. The Schema field value 'Name' is always required. The valid values for this field for the 'Blob' definition.objectType include 'Name, Creation-Time, Last-Modified, Content-Length, Content-MD5, BlobType, AccessTier, AccessTierChangeTime, Expiry-Time, hdi_isfolder, Owner, Group, Permissions, Acl, Snapshot, VersionId, IsCurrentVersion, Metadata, LastAccessedTime'. The valid values for 'Container' definition.objectType include 'Name, Last-Modified, Metadata, LeaseStatus, LeaseState, LeaseDuration, PublicAccess, HasImmutabilityPolicy, HasLegalHold'. Schema field values 'Expiry-Time, hdi_isfolder, Owner, Group, Permissions, Acl' are valid only for Hns enabled accounts."
+        }
+      },
+      "required": [
+        "format",
+        "schedule",
+        "objectType",
+        "schemaFields"
+      ],
+      "description": "An object that defines the blob inventory rule."
+    },
+    "BlobInventoryPolicyFilter": {
+      "properties": {
+        "prefixMatch": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "An array of strings for blob prefixes to be matched."
+        },
+        "blobTypes": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "An array of predefined enum values. Valid values include blockBlob, appendBlob, pageBlob. Hns accounts does not support pageBlobs. This field is required when definition.objectType property is set to 'Blob'."
```

Comment:
Per the description "This field is required when definition.objectType property is set to 'Blob'"
But we can add blob inventory policy with ObjectType as Blob, but no objectType property , like following.
So please make sure if the description incorrect, or server behavior incorrect?

```
DEBUG: ============================ HTTP REQUEST ============================

HTTP Method:
PUT

Absolute Uri:
https://management.azure.com/subscriptions/{subscription-id}/resourceGroups/***/providers/Microsoft.Storage/storageAccounts/***/inventoryPolicies/default?api-version=2021-04-01

Headers:
x-ms-client-request-id        : a842fecd-143b-488f-bfb1-f7ba83dd3ab7
accept-language               : en-US

Body:
{
  "properties": {
    "policy": {
      "enabled": false,
      "rules": [
        {
          "enabled": true,
          "name": "test2",
          "destination": "weitest1",
          "definition": {
            "format": "Parquet",
            "schedule": "Weekly",
            "objectType": "Blob",
            "schemaFields": [
              "Name",
              "Last-Modified",
              "Metadata",
              "LastAccessedTime"
            ]
          }
        }
      ],
      "type": "Inventory"
    }
  }
}


DEBUG: ============================ HTTP RESPONSE ============================

Status Code:
OK

Headers:
Pragma                        : no-cache
x-ms-request-id               : 8bda505f-de10-4cd8-a0f2-0bb223160659
Strict-Transport-Security     : max-age=31536000; includeSubDomains
x-ms-ratelimit-remaining-subscription-writes: 1199
x-ms-correlation-request-id   : fe0d7c84-d427-40e3-b6d2-1c79e2e09995
x-ms-routing-request-id       : SOUTHEASTASIA:20210427T065956Z:fe0d7c84-d427-40e3-b6d2-1c79e2e09995
X-Content-Type-Options        : nosniff
Cache-Control                 : no-cache
Date                          : Tue, 27 Apr 2021 06:59:55 GMT
Server                        : Microsoft-Azure-Storage-Resource-Provider/1.0,Microsoft-HTTPAPI/2.0 Microsoft-HTTPAPI/2.0

Body:
{
  "id": "/subscriptions/{subscription-id}/resourceGroups/***/providers/Microsoft.Storage/storageAccounts/***/inventoryPolicies/default",
  "name": "DefaultInventoryPolicy",
  "type": "Microsoft.Storage/storageAccounts/inventoryPolicies",
  "properties": {
    "lastModifiedTime": "2021-04-27T06:59:55.7238516Z",
    "policy": {
      "enabled": false,
      "type": "Inventory",
      "rules": [
        {
          "destination": "weitest1",
          "enabled": true,
          "name": "test2",
          "definition": {
            "format": "Parquet",
            "schedule": "Weekly",
            "objectType": "Blob",
            "schemaFields": [
              "Name",
              "Last-Modified",
              "Metadata",
              "LastAccessedTime"
            ]
          }
        }
      ]
    }
  }
}
```

Reply:
Thanks for pointing out, it is a server problem. blobTypes is always required for Blob object type and is not applicable for Container type. I have made the change and it will be deployed to stage region tomorrow.